RoboticonColony/Assets/Editor/Unit_Tests/MarketTest.cs

```diff
@@ -0,0 +1,173 @@
+﻿using UnityEngine;
+using UnityEditor;
+using NUnit.Framework;
+
+public class MarketTest {
+
+    [Test]
+    public void CreateMarketWithoutInventory()
+    {
+        //Create an empty market instance
+        Market market = new Market();
+
+        //Verify that the markets inventory is empty
+        Assert.AreEqual(0, market.Stock.Money);
+        Assert.AreEqual(0, market.Stock.GetItemAmount(ItemType.Ore));
+        Assert.AreEqual(0, market.Stock.GetItemAmount(ItemType.Power));
+        Assert.AreEqual(0, market.Stock.GetItemAmount(ItemType.Roboticon));
+    }
+
+	[Test]
+	public void CreateMarketWithInventory()
+	{
+        //Create an inventory to give to the market
+        Inventory inv = new Inventory(1, 4, 2, 3);
+
+        //Create the market instance
+        Market market = new Market(inv);
+
+        //Verify that the market contains the items from the inventory
+        Assert.AreEqual(1, market.Stock.Money);
+        Assert.AreEqual(4, market.Stock.GetItemAmount(ItemType.Ore));
+        Assert.AreEqual(2, market.Stock.GetItemAmount(ItemType.Power));
+        Assert.AreEqual(3, market.Stock.GetItemAmount(ItemType.Roboticon));
+    }
+
+    [Test]
+    public void GettingBuyPrices()
```

Comment:
maybe move the prices into the constructor so we can set them in the tests and check theyre the same here